src/mjolnir/shortcutbuilder.cc

```diff
@@ -406,6 +406,12 @@ uint32_t AddShortcutEdges(GraphReader& reader,
       DirectedEdge newedge = *directededge;
       uint32_t length = newedge.length();
 
+      //    if shortcut edge length is greater than kMaxEdgeLength, return 0
+      if (length > kMaxEdgeLength) {
+        LOG_ERROR("length too big");
+        throw std::runtime_error("edge length too big");
+      }
```

Comment:
hm, I had smth else in mind tbh. this just raises the same error as before, just with a slightly different error message, so to me it doesn't really add value yet. I think there's a fairly straight-forward way to handle this gracefully and still finish a valid graph build. what I thought we should do is detect when the max edge length is about to be exceeded, and if it would, we finish the current shortcut with a length less than the max and then create the next shortcut starting with the next edge. does that make sense?

Reply:
For a given shortcut, I understand that we want the total length of the shortcut to be shorter than the kMaxEdgeLength. When a shortcut is created, does the entire shortcut become one edge, or does it become a structure of connected edges?

Reply:
the entire shortcut will be a single edge. what we want to do, as @nilsnolde mentioned was to modify the `CanContract` method to return false if the contraction would make the final edge length longer than `kMaxEdgeLength`

Reply:
I made sure the CaContract returns false if the node's length is greater than the maxLength, but does the scope of the CanContract function allow us to keep track of the total shortcut length? It looks like it's being called before the shortcuts are built and does not know the length of the shortcut. I thought the length of the shortcuts could only be determined as FormShortcuts is called.

Reply:
I was speaking more to the semantics rather than how the code is today. Surely the call could have the total length passed in so far and bail if exceeded so that all the decision code is in one function. Or up one level we could track this decision separately. Either way will work I wasn't being specific

Reply:
Alright! Thank you for the clarification, that makes sense. I was wondering to what extent I could alter the code - looks like I have quite some freedom to work with!

Reply:
I've reflected on your advice, and added a condition to check for length in the AddShortcutEdges function. This seems to work, as tested on the test_shortcut file. I tested so that a shortcut is created between paths less than the max, but is not for edges greater than the cap.